image_tools/CMakeLists.txt

```diff
@@ -50,6 +53,56 @@ call_for_each_rmw_implementation(targets GENERATE_DEFAULT)
 if(BUILD_TESTING)
   find_package(ament_lint_auto REQUIRED)
   ament_lint_auto_find_test_dependencies()
+
+  # get the rmw implementations ahead of time
+  get_available_rmw_implementations(middleware_implementations)
+  foreach(middleware_impl ${middleware_implementations})
+    find_package("${middleware_impl}" REQUIRED)
+  endforeach()
+
+  # These are the regex's for validating the output of the executables.
+  set(RCLCPP_DEMO_SHOWIMAGE_EXPECTED_OUTPUT "${CMAKE_CURRENT_SOURCE_DIR}/test/showimage")
+  set(RCLCPP_DEMO_CAM2IMAGE_EXPECTED_OUTPUT "${CMAKE_CURRENT_SOURCE_DIR}/test/cam2image")
+
+  macro(testing_targets)
+    if(
+      # TODO(wjwwood): connext and connext_dynamic fail with:
+      #
+      #   COMMENDSrWriterService_write:!write. Reliable large data requires asynchronous writer.
+      #
+      # We can remove this disable for connext when that's fixed.
+      "${rmw_implementation} " STREQUAL "rmw_connext_cpp " OR
+      "${rmw_implementation} " STREQUAL "rmw_connext_dynamic_cpp " OR
+      # TODO(wjwwood): fastrtps fails with:
+      #
+      #   uncaught exception of type std::runtime_error: vector overcomes the maximum length
+      #
+      # We can remove this disable for fastrtps when that's fixed.
+      "${rmw_implementation} " STREQUAL "rmw_fastrtps_cpp "
+    )
+      message(STATUS "Skipping tests for '${rmw_implementation}'")
+      return()
```

Comment:
I assume the purpose of this return is to skip the current rmw implementation. But `return` doesn't work like that in a `macro`. Please see https://cmake.org/cmake/help/v3.5/command/return.html